### PR TITLE
refactor: fail fast when new NoExecuteTaintManager

### PR DESCRIPTION
**Pull request by kerthcet**
Signed-off-by: kerthcet <kerthcet@gmail.com>

#### What type of PR is this?
/kind cleanup
/sig apps

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
None
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/controller/nodelifecycle/scheduler/taint_manager.go [hunk 1]
```diff
@@ -156,15 +156,16 @@ func getMinTolerationTime(tolerations []v1.Toleration) time.Duration {
...
 	recorder := eventBroadcaster.NewRecorder(scheme.Scheme, v1.EventSource{Component: "taint-controller"})
 	eventBroadcaster.StartStructuredLogging(0)
-	if c != nil {
-		klog.V(0).InfoS("Sending events to api server")
-		eventBroadcaster.StartRecordingToSink(&v1core.EventSinkImpl{Interface: c.CoreV1().Events("")})
-	} else {
-		klog.Fatalf("kubeClient is nil when starting NodeController")
-	}
+
+	klog.V(0).InfoS("Sending events to api server")
```
Comment (VirrageS) [inline pkg/controller/nodelifecycle/scheduler/taint_manager.go:167]: I know that this is not related to your MR but I would suggest changing this to:
```suggestion
	klog.Info("Sending events to api server")
```

Comment (kerthcet) [reply pkg/controller/nodelifecycle/scheduler/taint_manager.go:167]: It doesn't make a lot of sense.

Comment (VirrageS) [reply pkg/controller/nodelifecycle/scheduler/taint_manager.go:167]: Why?

Comment (kerthcet) [reply pkg/controller/nodelifecycle/scheduler/taint_manager.go:167]: I think they behave the same, no performance improvement or better expression and it's better to keep the consistent style(we use `klog.V(x)` for the whole package). WDYT?

Comment (VirrageS) [reply pkg/controller/nodelifecycle/scheduler/taint_manager.go:167]: Ah, you are right. It seems like the `klog.V(x)` is used across this file (probably package as well, as you mentioned). Missed that.

Comment (VirrageS) [reply pkg/controller/nodelifecycle/scheduler/taint_manager.go:167]: So let's better leave it as is.

**Pull request closed without merging**